src/dsql/DdlNodes.epp

```diff
@@ -11998,7 +11998,12 @@ void AlterDatabaseNode::execute(thread_db* tdbb, DsqlCompilerScratch* dsqlScratc
 			Arg::PrivateDyn status(297);
 
 			if (conflict_trans)
-				status << Arg::Gds(isc_concurrent_transaction) << Arg::Num(conflict_trans);
+			{
+				// Cannot use Arg::Num here because transaction number is 64-bit signed integer
+				string trans_num_str;
```

Comment:
I suppose you've checked how does your code run.
Unfortunately here it runs due to some luck. status holds direct pointer to string argument till raise() is called but trans_num_str is destroyed before it. I.e. status has a pointer to already destroyed variable when raise is called. This may work sometimes but it's definitely a bug, you should move trans_num_str out of 'if' block.